node/actors/network/src/gossip/metrics.rs

```diff
@@ -0,0 +1,37 @@
+/// Metrics related to the gossiping of L1 batch votes.
+#[derive(Debug, vise::Metrics)]
+#[metrics(prefix = "network_gossip_batch_votes")]
+pub(crate) struct BatchVotesMetrics {
+    /// Number of members in the attester committee.
+    pub(crate) committee_size: vise::Gauge<usize>,
+
+    /// Number of votes added to the tally.
+    ///
+    /// Its rate of change should correlate with the attester committee size,
+    /// save for any new joiner casting their historic votes in a burst.
+    pub(crate) votes_added: vise::Counter,
+
+    /// Weight of votes added to the tally normalized by the total committee weight.
+    ///
+    /// Its rate of change should correlate with the attester committee weight and batch production rate,
+    /// that is, it should go up up by 1.0 with each new batch if everyone attests.
+    pub(crate) weight_added: vise::Counter<f64>,
+
+    /// The minimum batch number we still expect votes for.
+    ///
+    /// This should go up as the main node indicates the finalisation of batches,
+    /// or as soon as batch QCs are found and persisted.
+    pub(crate) min_batch_number: vise::Gauge<u64>,
+
+    /// Batch number in the last vote added to the register.
+    ///
+    /// This should go up as L1 batches are created, save for any temporary
+    /// outlier from lagging attesters or ones sending votes far in the future.
+    pub(crate) last_added_vote_batch_number: vise::Gauge<u64>,
+
+    /// Batch number of the last batch signed by this attester.
+    pub(crate) last_signed_batch_number: vise::Gauge<u64>,
```

Comment:
can postpone this, but can we have a `histogram` that would show the distribution of votes delay after an l1 batch was produced?

Reply:
Sounds interesting, I will create a ticket for this. There are a few challenges:
* the timestamp of the batch isn't available here
* currently we can receive votes earlier than we have the batch itself, so we would have to store a timestamp with the vote, and the retrospectively emit negative latencies when the batch is produced on the local node

We can have some specific workarounds: 
* For example when the `last_signed_batch_number` above changes it can act as an approximation of when the L1 batch was produced, but it's only an approximation because it depends on the polling frequency, and also it's only available on nodes which are attesters. 
* We could observe the time elapsed between the first and last vote. For attester nodes this would indicate the speed of propagation of votes over gossip, compared to the vote produced locally, although again the local vote might come later than the first gossiped one.
* We could implement Grzegorz's idea about only gossiping votes to nodes which have indicated they have the batch already, which would deal with the negative values; this simplifies the metrics in that we don't have to buffer timestamps and observe them the first time we learn about the batch.
* We could observe the difference between `l1_batches.created_at` and `l1_batches_consensus.created_at` when we insert the QC, which would be the overall latency of gossiping votes.

Reply:
https://linear.app/matterlabs/issue/BFT-497/add-metric-for-l1-batch-vote-latency